Mathlib/Algebra/BigOperators/Intervals.lean

```diff
@@ -203,10 +203,11 @@ theorem prod_Ico_id_eq_superFactorial : ∀ n : ℕ, (∏ x in Ico 1 (n + 1), x
       prod_Ico_id_eq_superFactorial n, superFactorial, factorial, Nat.succ_eq_add_one, mul_comm]
 
 @[simp]
-theorem prod_range_add_one_eq_superFactorial : ∀ n : ℕ, (∏ x in range n, (x + 1) !) = sf n
+theorem prod_range_add_one_eq_superFactorial : ∀ n : ℕ, (∏ x in range n, (x.succ) * x !) = sf n
```

Comment:
I wonder if there is some other way than chaning `(x + 1) !` to 'x.succ * x` to satisfy the simpNF linter..

Reply:
yes, that works. 
I will try what breaks if Nat.factorial_succ isn't a simp lemma...

Reply:
https://github.com/leanprover-community/mathlib4/pull/6840 seems like nothing breaks

Reply:
Even after #6840, we still need the `nolint`, since `factorial` still simplifies, because of https://github.com/leanprover-community/mathlib4/blob/f91160c3b897ef9698cd3706d2330c3b59fdb3e8/Mathlib/Data/Nat/Factorial/Basic.lean#L30

I considered removing the `simp` from the factorial here: #7078

However, it having the `simp` for the factorial seem to be useful in some cases.
What do you think, @alreadydone?

Reply:
I left a comment on #7078. I would prefer to make this PR (#6806) depend on that one (#7078).

Reply:
added dependency. Will make the needed changes one #7078 is resolved

Reply:
done